hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1307,23 +1296,6 @@ func validateAutoscaling(nodePool *hyperv1.NodePool) error {
 	return nil
 }
 
-func defaultNodePoolAMI(region string, releaseImage *releaseinfo.ReleaseImage) (string, error) {
-	// TODO: The architecture should be specified from the API
-	arch, foundArch := releaseImage.StreamMetadata.Architectures["x86_64"]
```

Comment:
Why deleting this? Let's keep this, keep using .Release as the source of truth to infer the default boot AMI and expose a field in the NodePool API to express intent for the Arch.
If the type can't be inferred then is required input.